File: engine/model/src/main/scala/tools/packet/monster/book/ChangeCover.scala
package tools.packet.monster.book

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ChangeCover(private var _cardId: Int) extends PacketInput {
  def cardId: Int = _cardId

  override def opcode(): SendOpcode = SendOpcode.MONSTER_BOOK_SET_COVER
}

File: engine/engine-model/src/main/java/tools/packet/monster/book/ChangeCover.java
package tools.packet.monster.book;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ChangeCover(Integer cardId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MONSTER_BOOK_SET_COVER;
   }
}